autogpt_platform/backend/backend/util/request.py

```diff
@@ -0,0 +1,128 @@
+import ipaddress
+import socket
+from typing import Callable
+from urllib.parse import urlparse
+
+import requests as req
+
+from backend.util.settings import Config
+
+# List of IP networks to block
+BLOCKED_IP_NETWORKS = [
+    ipaddress.ip_network("0.0.0.0/8"),  # "This" Network
+    ipaddress.ip_network("10.0.0.0/8"),  # Private-Use
+    ipaddress.ip_network("127.0.0.0/8"),  # Loopback
+    ipaddress.ip_network("169.254.0.0/16"),  # Link Local
+    ipaddress.ip_network("172.16.0.0/12"),  # Private-Use
+    ipaddress.ip_network("192.168.0.0/16"),  # Private-Use
+    ipaddress.ip_network("224.0.0.0/4"),  # Multicast
+    ipaddress.ip_network("240.0.0.0/4"),  # Reserved for Future Use
+]
```

Comment:
Hi @jackfromeast 

So I ended up needing to do this to reduce the impact of blocked IP ranges.
Domains like google.com mapped to a [special purpose ip](https://datatracker.ietf.org/doc/rfc6890/) like `192.0.0.88` which your initial code blocks. 

Do you think this will have a security repercussion?